src/main/java/com/itschool/eventmanagment/models/dtos/ParticipantDetailsDTO.java

```diff
@@ -0,0 +1,16 @@
+package com.itschool.eventmanagment.models.dtos;
+
+import lombok.Data;
+
+@Data
+public class ParticipantDetailsDTO {
+    private String firstName;
+    private String lastName;
+    private String email;
+
+    public ParticipantDetailsDTO(String firstName, String lastName, String email) {
```

Comment:
Does `@Data` annotation from Lombok provide a constructor already?

Reply:
Hi Alex, I have one error: "Incompatible parameter types in method reference expression".

"The error message "Incompatible parameter types in method reference expression" indicates that the types of arguments you're trying to pass to the constructor of ParticipantDetailsDTO using the method reference (ParticipantDetailsDTO::new) don't match the expected types. Even if you have @AllArgsConstructor (@Data), the field names and types in Participant and ParticipantDetailsDTO must be identical for the method reference to work."

 In my case, the field names and types in Participant and ParticipantDetailsDTO  are not the same.

Reply:
I think that is ok to generate my own constructor.

Reply:
You can try to use both @AllArgsConstructor and @NoArgsConstructor, I think. If this still doesn't work it's ok to have the custom one.